### PR TITLE
fix issue 1761 - TypeInfo.toString for function types always indicates no-parameter function

### PR DESCRIPTION
**Pull request by somzzz**
https://issues.dlang.org/show_bug.cgi?id=1761

Please review, thanks!

### CODE REVIEW

**Comment by andralex** [comment]
OK, good going. Now you may want to add a unittest covering the code you just added. That will make the codecov checks pass.

**Comment by andralex** [comment]
BTW, the codecov extension for Chrome will show the uncovered lines in dark red in the diff view.

File: src/object.d [hunk 1]
```diff
@@ -782,6 +782,21 @@ class TypeInfo_Function : TypeInfo
...
 }
 
+unittest
+{
+    class C
+    {
+       void func() {}
+       void func(int a) {}
+       int func(int a, int b) { return 0; }
+    }
```
Comment (andralex) [inline src/object.d]: I think you don't need to define these functions, just declare them - so your coverage won't be affected. Try it out.

```d
class C
{
        void func();
        void func(int a);
        int func(int a, int b);
}
```

Comment (somzzz) [inline src/object.d]: I made the class `C ` abstract and only declared the functions as you suggested. Is this better?

**Comment by andralex** [comment]
Great, thanks. The auto-tester has failed because of a timeout, I'll merge this.

**Pull request merged**